program-runtime/src/prioritization_fee.rs

```diff
@@ -0,0 +1,206 @@
+/// There are 10^6 micro-lamports in one lamport
+const MICRO_LAMPORTS_PER_LAMPORT: u64 = 1_000_000;
+
+type MicroLamports = u128;
+
+pub enum PrioritizationFeeType {
+    ComputeUnitPrice(u64),
+    Deprecated(u64),
+}
+
+#[derive(Default, Debug, PartialEq)]
+pub struct PrioritizationFeeDetails {
+    fee: u64,
+    priority: u64,
+}
+
+impl PrioritizationFeeDetails {
+    pub fn new(fee_type: PrioritizationFeeType, max_compute_units: u64) -> Self {
+        match fee_type {
+            PrioritizationFeeType::Deprecated(fee) => {
+                let priority = if max_compute_units == 0 {
+                    0
+                } else {
+                    let micro_lamport_fee: MicroLamports =
+                        (fee as u128).saturating_mul(MICRO_LAMPORTS_PER_LAMPORT as u128);
+                    let priority = micro_lamport_fee.saturating_div(max_compute_units as u128);
+                    u64::try_from(priority).unwrap_or(u64::MAX)
+                };
+
+                Self { fee, priority }
+            }
+            PrioritizationFeeType::ComputeUnitPrice(cu_price) => {
+                let fee = {
+                    let micro_lamport_fee: MicroLamports =
+                        (cu_price as u128).saturating_mul(max_compute_units as u128);
+                    let mut fee =
+                        micro_lamport_fee.saturating_div(MICRO_LAMPORTS_PER_LAMPORT as u128);
```

Comment:
I think you can also get this ceiling computation via

`(micro_lamport_fee + `MICRO_LAMPORTS_PER_LAMPORT` - 1).saturating_div(MICRO_LAMPORTS_PER_LAMPORT as u128)`